File: testdata/fixtures/kotlin-clean/src/main/kotlin/com/example/domain/Order.kt
// Domain layer: pure business types. Only the platform stdlib is allowed here.
package com.example.domain

import java.util.concurrent.atomic.AtomicLong

class Order(val total: Int) {
    val id: Long = NEXT_ID.getAndIncrement()

    override fun toString(): String = "order $id ($total)"

    companion object {
        private val NEXT_ID = AtomicLong(1)
    }
}

File: testdata/fixtures/scala-clean/src/main/scala/com/example/domain/Order.scala
// Domain layer: pure business types. Only the platform stdlib is allowed here.
package com.example.domain

import java.util.concurrent.atomic.AtomicLong

final class Order(val total: Int):
  val id: Long = Order.NextId.getAndIncrement()

  override def toString: String = s"order $id ($total)"

object Order:
  private val NextId = AtomicLong(1)
